### PR TITLE
Set GOPATH for both possible go guru execution paths

### PR DESCRIPTION
**Pull request by jblebrun**

### CODE REVIEW

File: autoload/go/guru.vim [hunk 1]
```diff
@@ -217,11 +213,19 @@ endfunc
 
 " run_guru runs the given guru argument
 function! s:run_guru(args) abort
+
```
Comment (fatih) [inline autoload/go/guru.vim]: Can you remove this newline please

File: autoload/go/guru.vim [hunk 2]
```diff
@@ -217,11 +213,19 @@ endfunc
...
+  let old_gopath = $GOPATH
+  let $GOPATH = go#path#Detect()
   if go#util#has_job()
-    return s:async_guru(a:args)
+    let abort = s:async_guru(a:args)
```
Comment (fatih) [inline autoload/go/guru.vim]: This has several meanings. For async it's the job id however for sync_guru it's the output. So the caller defines what to look. Can you rename it to `res` as for `result`? I think that would make more sense.

File: autoload/go/guru.vim [hunk 3]
```diff
@@ -217,11 +213,19 @@ endfunc
...
   endif
 
-  return s:sync_guru(a:args)
+  let abort = s:sync_guru(a:args)
+
+  let $GOPATH = old_gopath
+
+  return abort
+
```
Comment (fatih) [inline autoload/go/guru.vim]: Can you remove this newline as well please?

File: autoload/go/guru.vim (hunk 2 above)
Comment (jblebrun) [inline autoload/go/guru.vim]: I have no idea why I called it `abort`. I think I was distracted by the `abort` in the function signature. 

On closer examination... does the result need to be returned at all? No callers seem to use it. Can we just call an return nothing?

Comment (fatih) [inline autoload/go/guru.vim]: I was returning with the previous one, as we might need it in the future and it's handy. You can also just return however I'm not sure how to unset it easily (as Vim doesn't have defer like we have in Go)

Comment (jblebrun) [inline autoload/go/guru.vim]: Ok, I've left it and changed the name to `res` as you suggested.

File: autoload/go/guru.vim [hunk 4]
```diff
@@ -217,11 +213,17 @@ endfunc
...
 function! s:run_guru(args) abort
+  let old_gopath = $GOPATH
+  let $GOPATH = go#path#Detect()
   if go#util#has_job()
-    return s:async_guru(a:args)
+    let res = s:async_guru(a:args)
   endif
 
-  return s:sync_guru(a:args)
+  let res = s:sync_guru(a:args)
```
Comment (bhcleek) [inline autoload/go/guru.vim]: This line should now be in an else block so that guru isn't called twice when `async_guru` was called.

**Comment by fatih** [comment]
Thanks @jblebrun

**Pull request merged**